### PR TITLE
Sentry Logging Integration with Breadcrumb Hooks and Blacklist

### LINKED ISSUE #1006: Sentry is getting an ear-full
When we started our private testnet, the ethereum clients have been sending all `stdout` to `stderr`. This kills Sentry. We need to revive it.

### LINKED ISSUE #1068: Misconfigured Sentry LoggingIntegration
Sentry Issue: [NUCYPHER-14](https://sentry.io/organizations/nucypher/issues/1067418126/?referrer=github_integration)

### LINKED ISSUE #1070: w3.clientVersion returns a str, resulting in a KeyError from pubic chains
Sentry Issue: [NUCYPHER-15](https://sentry.io/organizations/nucypher/issues/1068549963/?referrer=github_integration)

```
KeyError: '5'
(11 additional frame(s) were not displayed)
...
  File "nucypher/config/node.py", line 331, in connect_to_blockchain
    sync=full_sync)
  File "nucypher/blockchain/eth/chains.py", line 141, in connect
    cls._instance.sync()
  File "nucypher/blockchain/eth/chains.py", line 98, in sync
    return self.interface.client.sync(timeout=timeout)
  File "nucypher/blockchain/eth/clients.py", line 201, in sync
    self.log.info(f"Waiting for {self.chain_name.capitalize()} chain synchronization to begin")
  File "nucypher/blockchain/eth/clients.py", line 146, in chain_name
    return PUBLIC_CHAINS[self.chain_id]
```

### PR DESCRIPTION
**Pull request by KPrasch**
- Update sentry DSN
- Adds sentry integration hooks
- Blacklists `NuCypherGethProcess` from the twisted logger flow
- Upgrades sentry and relocks project dependencies
- Allow `ethpm` installation on circleCI

Example: <https://github.com/nucypher/nucypher/pull/1069/files#diff-c45e89e41db078f139004005c2a8fa24R33>

Fixes #1068 
Fixes #1070 
Fixes #1006 

Related <https://github.com/nucypher/nucypher/issues/534>

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -3,85 +3,162 @@
 -e git+https://github.com/nucypher/py-solc.git@391b8da1a6bac5816877197bda25527c6b0b8c15#egg=py-solc
```
Comment (vepkenez) [inline dev-requirements.txt]: This file seems to have an unusually large amount of new requirements... what's up with that?

Comment (vepkenez) [reply dev-requirements.txt]: Are these duplicates of what's in requirements.txt?

File: nucypher/blockchain/eth/clients.py [hunk 2]
```diff
@@ -145,7 +143,7 @@ def peers(self):
     @property
     def chain_name(self) -> str:
         if not self.is_local:
-            return PUBLIC_CHAINS[self.chain_id]
+            return PUBLIC_CHAINS[int(self.chain_id)]
```
Comment (vepkenez) [inline nucypher/blockchain/eth/clients.py]: 👍  this has been bothering me!

File: nucypher/config/constants.py [hunk 3]
```diff
@@ -44,10 +44,9 @@
 SeednodeMetadata = namedtuple('seednode', ['checksum_address', 'rest_host', 'rest_port'])
 SEEDNODES = tuple()
 
-
 # Sentry
-NUCYPHER_SENTRY_PUBLIC_KEY = "d8af7c4d692e4692a455328a280d845e"
-NUCYPHER_SENTRY_USER_ID = '1310685'
+NUCYPHER_SENTRY_PUBLIC_KEY = "fd21c3edda324065a34d3f334dddf1f0"
```
Comment (vepkenez) [inline nucypher/config/constants.py]: In my opinion this should be 
```suggestion
NUCYPHER_SENTRY_PUBLIC_KEY = os.getenv("NUCYPHER_SENTRY_PUBLIC_KEY") or "fd21c3edda324065a34d3f334dddf1f0"
```

File: nucypher/config/constants.py [hunk 4]
```diff
@@ -44,10 +44,9 @@
...
+NUCYPHER_SENTRY_USER_ID = '1480080'
```
Comment (vepkenez) [inline nucypher/config/constants.py]: and this should be
```suggestion
NUCYPHER_SENTRY_USER_ID = os.getenv("NUCYPHER_SENTRY_USER_ID") or '1480080'
```

File: nucypher/blockchain/eth/clients.py (hunk 2 above)
Comment (KPrasch) [reply nucypher/blockchain/eth/clients.py]: Heh - yeah its little, see #1070

File: dev-requirements.txt (hunk 1 above)
Comment (KPrasch) [reply dev-requirements.txt]: Hmmm - checking. :thinking:

File: nucypher/config/constants.py (hunk 4 above)
Comment (KPrasch) [reply nucypher/config/constants.py]: Yeah - perhaps we need an Issue for standardizing where and how to define env vars - at the moment this check is done here: <https://github.com/nucypher/nucypher/pull/1069/files#diff-1ef88d57d518843bad3ce193d158843dR48>

Comment (KPrasch) [reply nucypher/config/constants.py]: See related #838

File: dev-requirements.txt (hunk 1 above)
Comment (KPrasch) [reply dev-requirements.txt]: Upon review, I think they look correct, although I am also wondering why there are so many new lines.

**Pull request merged**